### PR TITLE
Update Jenkins Integration Guide 

### PR DESCRIPTION
**Pull request by levlaz**
1. Update our integration template to make things more consistent 
2. Update Jenkins example to allow anyone to run it as is without needing to worry about having a specific type of project 
3. Fill out integration template for Jenkins 

There are a couple more thing that I want to do in the near future to improve this even further, but I think this PR is a good iteration that improves the existing Jenkins guide. 

Fixes docs-298

### CODE REVIEW

File: docs/current_docs/integrations/_integration.mdx.template [hunk 1]
```diff
@@ -2,16 +2,16 @@
 slug: /integrations/<name-of-integration>
 ---
 
-# <Name of Integration>
+# <name of integration>
```
Comment (jpadams) [inline docs/current_docs/integrations/_integration.mdx.template:5]: I wouldn't include an update to the template in this PR. Might be in by mistake?

File: docs/current_docs/integrations/jenkins.mdx [hunk 2]
```diff
@@ -4,9 +4,41 @@ slug: /integrations/jenkins
...
+Dagger provides a programmable container engine that allows you to replace your Groovy-based Jenkins pipelines with a regular programming language. This allows you to execute your pipeline the same locally and in Jenkins, with intelligent caching, while keeping all of your existing Jenkins infrastructure.
+
+# How it Works
+
+There are many ways to run Dagger on Jenkins. The simplest way to get started is having a Jenkins agent with OCI compatible container runtime.
+
+The general workflow looks like this:
+
+1. Jenkins receives job trigger as usual
+1. Jenkins loads Jenkinsfile from repo or from UI config
```
Comment (jpadams) [inline docs/current_docs/integrations/jenkins.mdx:16]:
```suggestion
1. Jenkins loads `Jenkinsfile` from repo or from UI config
```

File: docs/current_docs/integrations/jenkins.mdx [hunk 3]
```diff
@@ -4,9 +4,41 @@ slug: /integrations/jenkins
...
+1. Jenkins parses the Groovy based Jenkinsfile and finds `dagger call $foo`
```
Comment (jpadams) [inline docs/current_docs/integrations/jenkins.mdx:17]:
```suggestion
1. Jenkins parses the Groovy based `Jenkinsfile` and finds `dagger call $foo`
```

**Review by jpadams — APPROVED** [review]
Otherwise, looks great!
https://deploy-preview-7892--devel-docs-dagger-io.netlify.app/integrations/jenkins
Get it mergable and ship it! 🙏

File: docs/current_docs/integrations/jenkins.mdx [hunk 4]
```diff
@@ -4,9 +4,41 @@ slug: /integrations/jenkins
...
+1. Dagger Engine executes the pipeline called in `dagger call $foo` and sends telemetry to Dagger Cloud if the `DAGGER_CLOUD_TOKEN` environment variable is set.
+1. Job completes with success or failure, pipeline logs appear in Jenkins as usual
+
+# Prerequisites
+
+Running the examples shown below requires:
 
-This requires `docker` client installed on your Jenkins agent, a Docker host available (can be `docker:dind`), and agents labeled in Jenkins with `dagger`.
+1. At least one Jenkins agent with the `dagger` label
+1. A Docker host (can be `docker:dind`)
```
Comment (marcosnils) [inline docs/current_docs/integrations/jenkins.mdx:28]: "A Docker host" wasn't immediately clear to me when I first read it. Maybe reword it to make it more explicit what this is about?

File: docs/current_docs/integrations/snippets/Jenkinsfile [hunk 5]
```diff
@@ -4,17 +4,17 @@ pipeline {
   // assumes that the Dagger Cloud token
   // is in a Jenkins credential named DAGGER_CLOUD_TOKEN
   environment {
+    DAGGER_VERSION = "0.11.9"
```
Comment (marcosnils) [inline docs/current_docs/integrations/snippets/Jenkinsfile:7]: bump to v0.12.0?

**Review by marcosnils — COMMENTED** [review]
Hey Lev, arriving a bit late with comments :pray: LMK if my suggestions make sense and I can open a follow-up PR if necessary :pray:
